nexus/mgs-updates/src/test_util/mod.rs

```diff
@@ -0,0 +1,37 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use gateway_client::types::SpComponentCaboose;
+use hubtools::CabooseError;
+
+pub mod sp_test_state;
+pub mod step_through;
+pub mod test_artifacts;
+pub mod updates;
+
+/// Returns whether caboose `c1` (reported by an SP via MGS) matches caboose
+/// `c2` (constructed or read via `hubtools`)
+pub fn cabooses_equal(c1: &SpComponentCaboose, c2: &hubtools::Caboose) -> bool {
```

Comment:
Is it worth it to unpack one of these to ensure any fields added in the future get consideration?

```rust
// won't compile if new fields are added, at which point we may need
// to add extra checks below
let SpComponentCaboose { all, the, fields } = c1;
```

Reply:
Yes, definitely.  Doing this taught me that there's another field here: epoch.  I decided to ignore it because `hubtools` does not expose it and I don't think it affects the tests we have today.  But definitely worthwhile to do this for future-proofing.